### PR TITLE
NTO: Cloud provider specific TuneD profiles

### PR DESCRIPTION
**Pull request by jmencak**
Adding docs for the new 4.11 NTO functionality.

### CODE REVIEW

**Review by liqcui — COMMENTED** [review]
/lgtm

File: modules/custom-tuning-specification.adoc [hunk 1]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
 //     matchLabels:
 //       node-role.kubernetes.io/worker-custom: ""
 // EOF
+
+*Cloud provider specific TuneD profiles*
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:178]: provider-specific

File: modules/custom-tuning-specification.adoc [hunk 2]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+
+With this functionality, all Cloud provider specific nodes can conveniently
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:180]: provider-specific

File: modules/custom-tuning-specification.adoc [hunk 3]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+be assigned TuneD profile specifically tailored to a given Cloud provider
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:181]: a TundeD profile

File: modules/custom-tuning-specification.adoc [hunk 4]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+on {product-title} cluster without the need for adding any additional node
+labels or grouping nodes into Machine Config Pools.
+
+This functionality takes advantage of `spec.providerID` node object values
+in the form of `<cloud-provider>://<cloud-provider-specific-id>` and writes
+file `/var/lib/tuned/provider` with the value `<cloud-provider>` in NTO
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:187]: writes the file

File: modules/custom-tuning-specification.adoc [hunk 5]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+operand containers.  The content of this file is then used by TuneD to load
+`provider-<cloud-provider>` profile if such profile exists.
+
+The `openshift` profile that both `openshift-control-plane` and
+`openshift-node` profiles inherit settings from has been adjusted to take
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:192]: profiles inherit settings from is now updated to use this functionality through the use of conditional profile loading.

File: modules/custom-tuning-specification.adoc [hunk 6]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+advantage of this functionality by using conditional profile loading.
+Neither NTO nor TuneD currently ship any Cloud provider specific profiles.
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:194]: provider-specific

File: modules/custom-tuning-specification.adoc [hunk 7]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+However, it is possible to create a custom profile `provider-<cloud-provider>`
+that will be applied to all Cloud provider specific cluster nodes.
+
+.Example: GCE Cloud provider profile
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:198]: .Example GCE Cloud provider profile

File: modules/custom-tuning-specification.adoc [hunk 8]
```diff
@@ -174,3 +174,47 @@ To minimize node reboots, label the target nodes with a label the machine config
...
+      summary=GCE Cloud provider specific profile
+      # Your tuning for GCE Cloud provider goes here.
+    name: provider-gce
+----
+
+[NOTE]
+====
+Due to profile inheritance, any setting specified in
+`provider-<cloud-provider>` profile will be overwritten by the `openshift`
+and its child profiles.
```
Comment (ahardin-rh) [inline modules/custom-tuning-specification.adoc:219]: profile and its child profiles.

**Comment by jmencak** [comment]
Thank you for the review, @ahardin-rh .  Adjusted according to your suggestions.

[Jira story](https://issues.redhat.com/browse/PSAP-215) tracking this work.

**Pull request merged**